docs/source/reference-core.rst

```diff
@@ -677,6 +677,7 @@ nothing at all::
        with move_on_after(TIMEOUT):  # don't do this!
            nursery.start_soon(child)
 
+The reason for this is that the ``start_soon`` call immediately returns and the scheduled task actually runs inside the nursery. So in the above snippet the contextmanager ``with move_on_after(TIMEOUT):`` block is immediately exited after ``start_soon`` returns. Since ``move_on_after`` only imposes a time limit on the body of its ``with`` block and that block is exited it cannot cancel the child task.
```

Comment:
I expanded on this a bit further; feel free to apply my suggestion with or without changes if you think it makes all of this easier to understand! Also, please wrap lines in the documentation sources at approximately 80 chars -- we're not strict about this, but it's nice not to have things going way off the right hand side of the window when editing.

```suggestion
Why is this so? Well, ``start_soon()`` returns as soon as it has scheduled the new task to start running. The flow of execution in the parent then continues on to exit the ``with move_on_after(TIMEOUT):`` block, at which point Trio forgets about the timeout entirely. In order for the timeout to apply to the child task, Trio must be able to tell that its associated cancel scope will stay open for at least as long as the child task is executing. And Trio can only know that for sure if the cancel scope block is outside the nursery block.

You might wonder why Trio can't just remember "this task should be cancelled in ``TIMEOUT`` seconds", even after the ``with move_on_after(TIMEOUT):`` block is gone. The reason has to do with :ref:`how cancellation is implemented <cancellation>`. Recall that cancellation is represented by a `Cancelled` exception, which eventually needs to be caught by the cancel scope that caused it. (Otherwise, the exception would take down your whole program!) In order to be able to cancel the child tasks, the cancel scope has to be able to "see" the `Cancelled` exceptions that they raise -- and those exceptions come out of the ``async with open_nursery()`` block, not out of the call to ``start_soon()``.

If you want a timeout to apply to one task but not another, then you need to put the cancel scope in that individual task's function -- ``child()``, in this example.
```

Reply:
That is much much clearer then I could have managed. Thank you! I'm sure this will help future people running into the issue.